src/background_music/Loudness.h

```diff
@@ -0,0 +1,232 @@
+     * yields the peak window.
+     */
+    struct Spread
+    {
+
+        /**
+         * This returns seconds for step "step" of "of" steps.
+         */
+        static double get_seconds(size_t step, size_t of)
+        {
+            static constexpr double peak_perception_ratio =
+                Metrics::perception_peak_seconds /
+                Metrics::perception_center_seconds;
+            double log_ratio =
+                of > 0 ? static_cast<double>(std::clamp(step, size_t(0), of)) / static_cast<double>(of) : 1;
+            return Metrics::perception_center_seconds *
+                   pow(peak_perception_ratio, log_ratio);
+        }
+
+        /**
+         * This returns seconds and the weight for step "step" of "of" steps.
+         */
+        static void set_seconds_and_weight(double & seconds, double & weight,
+                                           size_t step, size_t of)
+        {
+            seconds = get_seconds(step, of);
+            weight = Metrics::get_weight(seconds);
+        }
+    };
+};
+
+class PerceptiveRMS
+{
+    class WindowedRMS
+    {
+        uint64_t window_sum_ = 0;
+        size_t window_size_ = 0;
+        size_t hold_samples_ = 0;
+        size_t hold_count_ = 0;
+        double window_multiplier_ = 1;
+        ScaledIntegrator<double> release_;
+
+    public:
+        [[nodiscard]] double get() const
+        {
+            return window_multiplier_ * release_.integrated();
+        }
+
+        double add_and_take_and_get(uint64_t add, uint64_t take)
+        {
+            window_sum_ += add;
+            window_sum_ -= take;
+            if (hold_samples_ && window_sum_ >= static_cast<uint64_t >(release_.integrated()))
+            {
+                hold_count_ = hold_samples_;
+                release_.set_value(static_cast<double>(window_sum_));
+            }
+            else if (hold_count_ > 0)
+            {
+                hold_count_--;
+            }
+            else
+            {
+                release_.integrate(static_cast<double>(window_sum_));
+            }
+            return get();
+        }
+
+        void set_multiplier(size_t window_size, double weight,
+                            size_t hold_samples)
+        {
+            window_size_ = window_size;
+            window_multiplier_ = window_size > 0 ? weight / static_cast<double>(window_size) : 1.0;
+            hold_samples_ = hold_samples;
+            hold_count_ = 0;
+            Integrator<double> i((double)hold_count_);
+            release_ = i;
+        }
+
+        void set_value(uint64_t value)
+        {
+            window_sum_ = static_cast<uint64_t >(std::round(static_cast<double>(value * window_sum_) / window_multiplier_));
+        }
+
+        [[nodiscard]] size_t window_size() const { return window_size_; }
+        [[nodiscard]] [[maybe_unused]] size_t hold_samples() const { return hold_samples_; }
+    };
+
+    RingBuf<uint64_t> buffer_;
+    WindowedRMS * rms_;
```

Comment:
Please use a suitable container (`Index` or `std::vector` or similar) to manage ownership rather than a raw pointer.